daml/splice-dso-governance/daml/Splice/DsoRules.daml

```diff
@@ -1297,10 +1297,11 @@ template DsoRules with
         let numValidatorFaucetCoupons = optional 0.0 (intToDecimal . length) optValidatorFaucetCouponCids
 
         -- expire validator liveness activity records and compute their unclaimed rewards
-        forA_ (fromOptional [] optValidatorLivenessActivityRecordCids) $ \validatorLivenessActivityRecordCid -> do
-          exercise validatorLivenessActivityRecordCid  ValidatorLivenessActivityRecord_DsoExpire with closedRoundCid
+        expiredValidatorLivenessActivityRecords <- forA (fromOptional [] optValidatorLivenessActivityRecordCids) $ \validatorLivenessActivityRecordCid -> do
+          expireResult <- exercise validatorLivenessActivityRecordCid  ValidatorLivenessActivityRecord_DsoExpire_V2 with closedRoundCid
+          return expireResult.weight
```

Comment:
I know the original code is alrady duplicated. However, if we go for the `fetch` solution, then it would be quite nice if we shared that code between `AmuletRules` and `DsoRules`.